src/main/java/org/opensearch/ad/rest/RestStatsAnomalyDetectorAction.java

```diff
@@ -11,64 +11,119 @@
 import org.opensearch.ad.settings.EnabledSetting;
 import org.opensearch.ad.stats.ADStats;
 import org.opensearch.ad.transport.ADStatsRequest;
 import org.opensearch.ad.transport.StatsAnomalyDetectorAction;
+import org.opensearch.ad.transport.StatsAnomalyDetectorResponse;
 import org.opensearch.ad.util.DiscoveryNodeFilterer;
-import org.opensearch.client.node.NodeClient;
+import org.opensearch.ad.util.RestHandlerUtils;
 import org.opensearch.cluster.node.DiscoveryNode;
-import org.opensearch.cluster.service.ClusterService;
 import org.opensearch.common.Strings;
-import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.json.JsonXContent;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.extensions.rest.ExtensionRestResponse;
 import org.opensearch.rest.RestRequest;
-import org.opensearch.rest.action.RestToXContentListener;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.sdk.BaseExtensionRestHandler;
+import org.opensearch.sdk.ExtensionsRunner;
+import org.opensearch.sdk.RouteHandler;
+import org.opensearch.sdk.SDKClient.SDKRestClient;
 
 import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
 
 /**
- * RestStatsAnomalyDetectorAction consists of the REST handler to get the stats from the anomaly detector plugin.
+ * RestStatsAnomalyDetectorAction consists of the REST handler to get the stats from the anomaly detector extension.
  */
-public class RestStatsAnomalyDetectorAction extends BaseRestHandler {
+public class RestStatsAnomalyDetectorAction extends BaseExtensionRestHandler {
 
     private static final String STATS_ANOMALY_DETECTOR_ACTION = "stats_anomaly_detector";
+    private final Logger logger = LogManager.getLogger(RestStatsAnomalyDetectorAction.class);
+    @Inject
     private ADStats adStats;
-    private ClusterService clusterService;
     private DiscoveryNodeFilterer nodeFilter;
-
-    /**
-     * Constructor
-     *
-     * @param adStats ADStats object
-     * @param nodeFilter util class to get eligible data nodes
-     */
-    public RestStatsAnomalyDetectorAction(ADStats adStats, DiscoveryNodeFilterer nodeFilter) {
-        this.adStats = adStats;
-        this.nodeFilter = nodeFilter;
+    private SDKRestClient sdkRestClient;
+    private Settings settings;
+    @Inject
+    private DiscoveryNodeFilterer discoveryNodeFilterer;
+
+    public RestStatsAnomalyDetectorAction(ExtensionsRunner extensionsRunner, SDKRestClient sdkRestClient) {
+        this.sdkRestClient = sdkRestClient;
+        this.settings = extensionsRunner.getEnvironmentSettings();
     }
 
-    @Override
     public String getName() {
         return STATS_ANOMALY_DETECTOR_ACTION;
     }
 
     @Override
-    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) {
+    public List<RouteHandler> routeHandlers() {
```

Comment:
This implementation works for now and no need to hold up this PR.  But heads up that with [SDK #665](https://github.com/opensearch-project/opensearch-sdk-java/issues/665) we'll be able to go back to the replaced routs format of existing rest handler classes.  So stay tuned!